deepstream/5.1/deepstream_ssd_parser.py

```diff
@@ -0,0 +1,518 @@
+#!/usr/bin/env python3
+
+################################################################################
+# Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+#
+# Permission is hereby granted, free of charge, to any person obtaining a
+# copy of this software and associated documentation files (the "Software"),
+# to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense,
+# and/or sell copies of the Software, and to permit persons to whom the
+# Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+# DEALINGS IN THE SOFTWARE.
+################################################################################
+
+""" Example of deepstream using SSD neural network and parsing SSD's outputs. """
+
+import sys
+import io
+sys.path.append("../")
+import gi
+gi.require_version("Gst", "1.0")
+from gi.repository import GObject, Gst
+from common.is_aarch_64 import is_aarch64
+from common.bus_call import bus_call
+from ssd_parser import nvds_infer_parse_custom_tf_ssd, DetectionParam, NmsParam, BoxSizeParam
+import pyds
+from common.FPS import GETFPS
+
+fps_streams={}
+CLASS_NB = 91
+ACCURACY_ALL_CLASS = 0.5
+UNTRACKED_OBJECT_ID = 0xffffffffffffffff
+IMAGE_HEIGHT = 1080
+IMAGE_WIDTH = 1920
+MIN_BOX_WIDTH = 32
+MIN_BOX_HEIGHT = 32
+TOP_K = 20
+IOU_THRESHOLD = 0.3
+OUTPUT_VIDEO_NAME = "./out.mp4"
```

Comment:
Based on provided `output_directory` and the file name you can create an unique address for the output video path:
for example if the user provides `/home/deepstream_output/` as the output directory and the input video file is `softbio.mp4` the `OUTPUT_VIDEO_NAME` will be `/home/deepstream_output/softbio_neuralet_deepstream.mp4`

Reply:
done.